cli/src/execDocker/docker.js

```diff
@@ -116,8 +141,15 @@ export async function pushDockerImage({
       password: dockerhubAccessToken,
     },
   });
-
   await new Promise((resolve, reject) => {
+    // Handle abort signal
+    if (signal) {
+      signal.addEventListener('abort', () => {
+        imagePushStream.destroy();
+        reject(new Error('Docker push process was unexpectedly terminated.'));
+      });
+    }
```

Comment:
same here simply pass the `abortSignal` option to `docker.push()` and let `dockerode` do its stuff